rest/rest-api/src/main/java/org/apache/brooklyn/rest/api/AdjunctApi.java

```diff
@@ -253,4 +253,12 @@ public Response setConfig(
             @ApiParam(value = "Whether to include subtasks recursively across different entities (default false)", required = false)
             @QueryParam("recurse") @DefaultValue("false") Boolean recurse);
 
+    @GET
+    @ApiOperation(value = "Returns an icon for the adjunct, if defined")
+    @Path("/{adjunct}/icon")
+    public Response getIcon(
+        @ApiParam(value = "Application ID or name", required = true) @PathParam("application") String applicationId,
+        @ApiParam(value = "Entity ID or name", required = true) @PathParam("entity") String entityId,
+        @ApiParam(value = "Adjunct ID or name", required = true) @PathParam("adjunct") String adjunctToken);
+
```

Comment:
Is it necessary to have this endpoint if the icon is returned within the `AdjunctSummary`?

Reply:
it's only the `iconUrl` returned in adjunct summary.  the url might be a server-side `classpath:`.  this endpoint returns the actual image.

Reply:
@ahgittin Hum, so I just checked and the way the `applications` and `entities` endpoints work is as follow: the `iconUrl` field returned by the `EntitySummary` is a relative link to the related icon endpoint, i.e. `/v1/entities/icon/symbolicName/version`

So indeed, we need this icons endpoint for adjunct. But more importantly, we should mirror the same behaviour.

Reply:
@tbouron what are you looking at?  `/v1/entities/icon/symbolicName/version` isn't a path i'm familiar with.

i've based this on `EntityApi.getIcon` which corresponds to `/v1/applications/XXX/entities/XXX/icon` exactly as does this.

`CatalogResource.getIcon` is different but we should mirror entity instances.  (you could also use the `/v1/catalog/icon/XXX/VVV` endpoint for this but that won't look at icon tags on the instances.)

Reply:
@ahgittin Sorry, I meant `iconUrl: /v1/catalog/icon/symbolicName/version`.

For example, `/v1/applications` returns the above `iconUrl` if the catalog items have icons

Reply:
as we're dealing with adjunct instances here best to copy the entity instance api as i've done, no?

whether an entity returns `/v1/applications/XXX/entities/XXX/icon` or `/v1/catalog/icon/symbolicName/version` as the URL is not guaranteed.  in fact the latter would sometimes be wrong if an entity has a custom icon set in a tag, and the code suggests entity instances returns the former, at least for `EntityDetail`, in `ApplicationResource.fromEntity`:

    iconUrl = EntityTransformer.entityUri(entity, ui.getBaseUriBuilder())+"/icon";

the `EntitySummary` class doesn't include `iconUrl`.  you probably mean `CatalogEntitySummary` which is for types not instances in which case the non-instance link would be the correct one.

but for this purpose -- adjunct _instances_ - i think what i've done, mirroring entity instances, is correct.

Reply:
Yeah, I probably mixed up the name of REST endpoints with the multiple methods of getting an icon. There are too many of them, it's really confusing and hard to follow our REST API!

This looks fine to me.

Reply:
Thx.

We should introduce `/v1/catalog/type/XXX/V/icon` for consistency with `/v1/applications/XXX/icon` and `/v1/applications/XXX/entities/XXX/icon` and `/v1/applications/XXX/entities/XXX/adjuncts/XXX/icon` and deprecate /v1/catalog/icon/XXX/V`.  But I think our multiple methods are sensible, you can get an icon for instances, and for a type.

removing deprecated items like `CatalogEntitySummary` and others in favour of `TypeSummary` should make it simpler

think on any other confusing things and let me know